src/core/bpf-firewall.c

```diff
@@ -392,9 +430,22 @@ static int bpf_firewall_prepare_access_maps(
                 if (!cc)
                         continue;
 
-                bpf_firewall_count_access_items(verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny, &n_ipv4, &n_ipv6);
+                list = verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny;
+
+                bpf_firewall_count_access_items(list, &n_ipv4, &n_ipv6);
+
+                if (!LIST_IS_EMPTY(list) && n_ipv4 == 1 && n_ipv6 == 1 && !a)
```

Comment:
most of this check should be done inside of ip_address_access_item_is_any() i think